src/test/scala/io/iohk/ethereum/SecureRandomProvider.scala

```diff
@@ -0,0 +1,11 @@
+package io.iohk.ethereum
+
+import java.security.SecureRandom
+
+import io.iohk.ethereum.utils.Config
+
+trait SecureRandomProvider {
+  val secureRandom: SecureRandom = SecureRandom.getInstance(Config.secureRandomAlgo)
+}
+
+object SecureRandomProvider extends SecureRandomProvider
```

Comment:
`SecureRandomProvider` in test shadows `SecureRandomProvider` without linking the types. What we want is to use the type in `main` in the test area as we would building another component for test support. Then if it's required, override it using `override val secureRandom` in tests, or if it's useful across many tests use a trait (eg) `TestRandomProvider extends SecureRandomProvider`. Even if we want to shadow component builders with more test oriented ones, they have to link to the original builders, otherwise it's going to get very confusing.